src/MEVCapturingHook.sol

```diff
@@ -25,14 +43,56 @@ contract MEVCapturingHook is BaseHook {
             beforeRemoveLiquidity: false,
             afterAddLiquidity: false,
             afterRemoveLiquidity: false,
-            beforeSwap: false,
+            beforeSwap: true,
             afterSwap: false,
             beforeDonate: false,
             afterDonate: false,
-            beforeSwapReturnDelta: false,
+            beforeSwapReturnDelta: true,
             afterSwapReturnDelta: false,
             afterAddLiquidityReturnDelta: false,
             afterRemoveLiquidityReturnDelta: false
         });
     }
+
+    function _beforeInitialize(address, PoolKey calldata key, uint160) internal override returns (bytes4) {
+        if (!key.fee.isDynamicFee()) revert MustUseDynamicFee();
+
+        poolConfig[key.toId()] = PoolConfig(DEFAULT_FEE_UNIT, DEFAULT_PRIORITY_THRESHOLD);
+
+        return this.beforeInitialize.selector;
+    }
+
+    function _beforeSwap(address, PoolKey calldata key, IPoolManager.SwapParams calldata params, bytes calldata)
+        internal
+        override
+        returns (bytes4, BeforeSwapDelta, uint24)
+    {
+        // take a fee based on the priority fee
+        // and donate it to LP
+
+        uint256 priorityFee = _getPriorityFee();
+
+        if (priorityFee < poolConfig[key.toId()].priorityThreshold || block.number == lastTradedBlock[key.toId()]) {
```

Comment:
u are using severl times key.toId(). Would it be more gas efficient to store it once in a  local variable